woocommerce/woocommerce
# Add: Active Filters block powered by Interactivity API

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





Closes #42179  .



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Add the `Collection Filters` block into a `Product Collection` block.
2. See the `Collection Active Filters` and other filter blocks added along with the `Collection Filters` block.
3. Toggle the display style, see the editor reacts.
4. Save and test the block on the frontend:
  - It's hidden when no filter applied.
  - It's visible when a filter is applied.
  - Clicking on the remove button of each active filter removes that filter from the URL and update the products grid as well as the corresponding filter block.
  - Clicking on the `Clear All` button remove all filters.
  - Removing a filter from the filter block also remove the corresponding one from the active filter block.
  - See chip and list style work as expected.

Note that the price filter nêd to be updated to use `context` instead of `state` to make it work properly with `Clear All` button.



### Changelog entry




<details>

#### Significance




#### Type




#### Message 

#### Comment 
Add new Active Filters block powered by Interactivity API.
</details>
